markdowns/02_Finetune_a_model_on_your_data.md

```diff
@@ -1,161 +1,113 @@
 
-```python
-# Saving the model happens automatically at the end of training into the `save_dir` you specified
-# However, you could also save a reader manually again via:
-reader.save(directory="my_model")
-```
+2. Provide the SQuAD format training data to the `Reader.train()` method.
 
 
 ```python
-# If you want to load it at a later point, just do:
-new_reader = FARMReader(model_name_or_path="my_model")
+data_dir = "data/squad20"
+reader.train(
+    data_dir=data_dir,
+    train_filename="dev-v2.0.json",
+    use_gpu=True,
+    n_epochs=1,
+    save_dir="my_model"
+)
 ```
 
-## Distill your model
-In this case, we have used "distilbert-base-uncased" as our base model. This model was trained using a process called distillation. In this process, a bigger model is trained first and is used to train a smaller model which increases its accuracy. This is why "distilbert-base-uncased" can achieve quite competitive performance while being very small.
-
-Sometimes, however, you can't use an already distilled model and have to distil it yourself. For this case, haystack has implemented [distillation features](https://haystack.deepset.ai/guides/model-distillation).
-
-### Augmenting your training data
-To get the most out of model distillation, we recommend increasing the size of your training data by using data augmentation. You can do this by running the [`augment_squad.py` script](https://github.com/deepset-ai/haystack/blob/main/haystack/utils/augment_squad.py):
-
+With the default parameters above, we are starting with a base model trained on the SQuAD training dataset and we are further fine-tuning it on the SQuAD development dataset. To fine-tune the model for your domain, replace `train_filename` with your domain-specific dataset.
 
-```python
-# Downloading script
-!wget https://raw.githubusercontent.com/deepset-ai/haystack/main/haystack/utils/augment_squad.py
+To perform evaluation over the course of fine-tuning, see [FARMReader.train() API](https://docs.haystack.deepset.ai/reference/reader-api#farmreadertrain) for the relevant arguments.
 
-doc_dir = "data/tutorial2"
+## Saving and Loading
 
-# Downloading smaller glove vector file (only for demonstration purposes)
-glove_url = "https://nlp.stanford.edu/data/glove.6B.zip"
-fetch_archive_from_http(url=glove_url, output_dir=doc_dir)
+The model is automatically saved at the end of fine-tuning in the `save_dir` that you specified.
+However, you can also manually save the Reader again by running:
 
-# Downloading very small dataset to make tutorial faster (please use a bigger dataset for real use cases)
-s3_url = "https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/squad_small.json.zip"
```

Comment:
At this point I am baffled and I think I need help figuring this out @julian-risch 
Right now, we look at this dataset to count the times tutorial 2 has been run. Now we're decoupling the distillation bit so this line is out. However, we don't at any point for the training step have users download the data/squad20 "train_filename="dev-v2.0.json"? But the tutorial works. Does the `train()` function pull it by name from somewhere? Is it possible to once we merge these tutorial restructures that we track that? @agnieszka-m tagging you here for visibility

Reply:
That can't work. Only possible explanation is that you already have the `data_dir = "data/squad20"`directory with some data in it. For example if you ran another tutorial before. We need to also fetch_archive_from_http the data in tutorial 2. So this needs to be added here again. And I could upload a/ datasets/documents/squad_small2.json.zip" or something like that with a different dataset name than before for telemetry.

Reply:
Ah, I'm wrong.
I just tried running the new tutorial here: https://colab.research.google.com/github/deepset-ai/haystack-tutorials/blob/tutorial_restructure_draft/tutorials/02_finetune_a_reader.ipynb and it does download the dataset.
The explanation is that the processor has two datasets hardcoded here: https://github.com/deepset-ai/haystack/blob/e266cf6e29f78df751d9dbe7a505886579233aa5/haystack/modeling/data_handler/processor.py#L42 One is squad

Here is the logic that decides about downloading that data: https://github.com/deepset-ai/haystack/blob/e266cf6e29f78df751d9dbe7a505886579233aa5/haystack/modeling/data_handler/processor.py#L2250

Reply:
Thanks, in that case for now, until we have any decisions to change or keep the way we do telemetry for tutorials, I will add a `fetch_archive_from_http` to tutorial 2 as well. As you suggested in the comment above. I will create a list of datasets for tutorials in the meantime. Until and if there is a change, it will be healthier to have.

I've already decoupled tutorial 1 and 3 into a separate PR as there's no need for those 2 to wait while we're fixing datasets etc for Tutorial 2 and 21..